test/test-clusters/src/main/java/org/elasticsearch/test/cluster/local/LocalClusterSpec.java

```diff
@@ -103,6 +105,7 @@ public static class LocalNodeSpec {
         private final List<SystemPropertyProvider> systemPropertyProviders;
         private final Map<String, String> systemProperties;
         private final List<String> jvmArgs;
+        private final Supplier<Path> configDirSupplier;
```

Comment:
FWIW, I don't think we need to plumb laziness all the way through to the spec here. We only build the spec once we attempt to start the server, at which point everything we depend on needs to be initialized anyway. Let's just make this a `Path` to make downstream usages simpler.

Reply:
Makes sense. Pushed 0651da01e6d306352c71363e41023450ecf9e69c